projects/ngx-maplibre-gl/src/lib/source/source.directive.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DestroyRef,
+  Directive,
+  OnInit,
+  inject,
+  input,
+  signal,
+} from '@angular/core';
+import { Subject, fromEvent } from 'rxjs';
+import { filter, switchMap, tap } from 'rxjs/operators';
+import { MapService } from '../map/map.service';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Source, SourceSpecification } from 'maplibre-gl';
+
+/**
+ * `mgl-canvas-source` - a canvas source component
+ * @see [canvas](https://maplibre.org/maplibre-style-spec/sources/#canvas)
+ *
+ * @category Source Components
+ */
+@Directive({
```

Comment:
Does this helper has to be a directive?

Reply:
Yes, its used as hostDirective in other component